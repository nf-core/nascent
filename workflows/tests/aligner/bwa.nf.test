nextflow_pipeline {

    name "BWA"
    script "../../../main.nf"
    tag "aligner"
    tag "bwa"

    test("Should run with defaults") {

        when {
            params {
                outdir = "$outputDir"
                aligner = "bwa"
                input = "${projectDir}/assets/samplesheet.csv"
            }
        }

        then {
            // stable_name: All files + folders in ${params.outdir}/ with a stable name
            def stable_name = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore: ['pipeline_info/*.{html,json,txt}', 'multiqc/multiqc_plots', 'multiqc/multiqc_plots/**'])
            // stable_path: All files in ${params.outdir}/ with stable content
            def stable_path = getAllFilesFromDir(params.outdir, ignoreFile: 'tests/.nftignore')
            // bam_files: All bam files
            def bam_files = getAllFilesFromDir(params.outdir, include: ['**/*.bam'])
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    // Number of tasks
                    workflow.trace.succeeded().size(),
                    // pipeline versions.yml file for multiqc from which Nextflow version is removed because we tests pipelines on multiple Nextflow versions
                    removeNextflowVersion("$outputDir/pipeline_info/nf_core_pipeline_software_mqc_versions.yml"),
                    // All stable path name
                    stable_name,
                    // All files with stable contents
                    stable_path,
                    // All bam files
                    // FIXME bam_files.collect{ file -> [ file.getName(), bam(file.toString()).getHeaderMD5() ] }
                ).match() }
            )
        }
    }

    def igenomes_base = 's3://ngi-igenomes/igenomes'

    // https://github.com/nf-core/nascent/issues/106
    test("Should work with BWA Index") {
        // FIXME PINTS Fails because it doesn't find anything.
        config '../skip_pints.config'
        when {
            params {
                outdir = "$outputDir"
                aligner = "bwa"
                // TODO Update these to human
                bwa_index = "${igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BWAIndex/version0.6.0/"
                fasta = "${igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/WholeGenomeFasta/genome.fa"
                gtf = "${igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Annotation/Genes/genes.gtf"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.trace.tasks().size(),
                ).match()
                }
            )
        }
    }

    // https://github.com/nf-core/nascent/issues/119
    test("Should work with gzipped references") {
        config '../skip_gzip.config'
        when {
            params {
                outdir = "$outputDir"
                aligner = "bwa"
                fasta = 'https://raw.githubusercontent.com/nf-core/test-datasets/rnaseq/reference/genome.fasta.gz'
                gtf = 'https://raw.githubusercontent.com/nf-core/test-datasets/rnaseq/reference/genes.gtf.gz'
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.trace.tasks().size(),
                ).match()
                }
            )
        }
    }
}
