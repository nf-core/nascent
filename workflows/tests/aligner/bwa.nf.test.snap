{
    "output_files": {
        "content": [
            149,
            [
                "cd4_REP1.trimmed.fastp.json:md5,bd9a3344c1591d6be4d524451f9dca53",
                "cd4_REP2.trimmed.fastp.json:md5,bc6e3f9ff7835f220535cc393b8eb25f",
                "cd4_REP3.trimmed.fastp.json:md5,c9c3a1dd0a2ebac77c9d7c3b0daf3790",
                "cd4_REP4.trimmed.fastp.json:md5,c073bd26609ddca91396d0f374ec5b80",
                "jurkat_REP1.trimmed.fastp.json:md5,329fd8e56421a196fc2ebd37ab3bc22d",
                "jurkat_REP2.trimmed.fastp.json:md5,8bad5b02a27e5386255b998a6a69db15"
            ],
            "c529a16c839e85e119b98354f109352d",
            "9e63b682af88fa902cf92b5c485845b1",
            "2aa200bbbce374ec7dedb9b46ce4aae1",
            "122274730dd5386e1b26c174df886a8b",
            "c2adf5327ff6d4edda2fdad00c7cb9bf",
            "63fea3b1e2e08e5f6fdf3f7ba0aa7818",
            [
                "cd4_REP1.sorted.bam.flagstat:md5,863e2d506d5cc4239af98a5f31bbc906",
                "cd4_REP1.sorted.bam.idxstats:md5,b1dd8bcbd23c53c21f0e11082d9315f2",
                "cd4_REP1.sorted.bam.stats:md5,1536c80bae78b2062508e1de210f6387",
                "cd4_REP2.sorted.bam.flagstat:md5,0fd86dbf8f799fad49ba471702979bdc",
                "cd4_REP2.sorted.bam.idxstats:md5,53204e4c6a9f68664087e4a8123be46a",
                "cd4_REP2.sorted.bam.stats:md5,561610e53fb676ac83252712dcac30d4",
                "cd4_REP3.sorted.bam.flagstat:md5,793c9cadbf80e35dbae3774e9b5bd93f",
                "cd4_REP3.sorted.bam.idxstats:md5,42de227ff397013cf288759871b0e938",
                "cd4_REP3.sorted.bam.stats:md5,9c8a44ba139ce9a24449779f5d3c9228",
                "cd4_REP4.sorted.bam.flagstat:md5,ed6c609505097f3cc9d9902d69e016f8",
                "cd4_REP4.sorted.bam.idxstats:md5,959dc185ae59de4cebe374026711a55e",
                "cd4_REP4.sorted.bam.stats:md5,edc57478278aa35b7376cee9afd634a7",
                "jurkat_REP1.sorted.bam.flagstat:md5,fd5f02b0f02a407447b804b1d80f5421",
                "jurkat_REP1.sorted.bam.idxstats:md5,c61af0847c1ad76c06a8de2815975b32",
                "jurkat_REP1.sorted.bam.stats:md5,ce9ff516908d86fb6db6ce44d5a3904b",
                "jurkat_REP2.sorted.bam.flagstat:md5,86ed47bd41a745ab59de473082c7742d",
                "jurkat_REP2.sorted.bam.idxstats:md5,3db8f88c1f836eb5d10f37ca1df81ae9",
                "jurkat_REP2.sorted.bam.stats:md5,026c558e8c36136ef19bfbef6b12ce2b"
            ],
            [
                "cd4_REP1.coverage.hist.txt:md5,7cbb473be8d3b32ff2e52fdf4e5d10d2",
                "cd4_REP1.coverage.stats.txt:md5,f1471b61ac17dba283d80e08450c7e55",
                "cd4_REP2.coverage.hist.txt:md5,8d5258a0882494bc4e3f1aa6aa5ed685",
                "cd4_REP2.coverage.stats.txt:md5,3cdb4473211f9da44166ffa6aaa5b602",
                "cd4_REP3.coverage.hist.txt:md5,4a8397c1ea08f35ca7046e3c2c014a08",
                "cd4_REP3.coverage.stats.txt:md5,b88592b469d504a93f2389255ac15f38",
                "cd4_REP4.coverage.hist.txt:md5,d2cbf89e1f753d99bcd03b4e472270dd",
                "cd4_REP4.coverage.stats.txt:md5,4c2b5fe6793aab8560cc87c176ac06ab",
                "jurkat_REP1.coverage.hist.txt:md5,71a893d9d1d55fd47399f6d47b628d6e",
                "jurkat_REP1.coverage.stats.txt:md5,381c69a30099d82066a959deab1a2569",
                "jurkat_REP2.coverage.hist.txt:md5,d124bdadaf2dc5cdd9e3db5a1ab4c588",
                "jurkat_REP2.coverage.stats.txt:md5,b1940346719e4070091f2eef6e508f49"
            ],
            [
                "cd4_REP1.c_curve.txt:md5,cf4743abdd355595d6ec1fb3f38e66e5",
                "cd4_REP1.lc_extrap.txt:md5,8633f84ccd5cc725db9af4b33edd63b0",
                "cd4_REP2.c_curve.txt:md5,cf4743abdd355595d6ec1fb3f38e66e5",
                "cd4_REP2.lc_extrap.txt:md5,3ad9e4028c3711e6d46831c10ed04200",
                "cd4_REP3.c_curve.txt:md5,cf4743abdd355595d6ec1fb3f38e66e5",
                "cd4_REP3.lc_extrap.txt:md5,25ebbeabee46caa86a4d6d82dd0091f1",
                "cd4_REP4.c_curve.txt:md5,cf4743abdd355595d6ec1fb3f38e66e5",
                "cd4_REP4.lc_extrap.txt:md5,d0df86774dd54421fd362d46faa3f937",
                "jurkat_REP1.c_curve.txt:md5,cf4743abdd355595d6ec1fb3f38e66e5",
                "jurkat_REP1.lc_extrap.txt:md5,84faa937faa88476c942f330c5762cb5",
                "jurkat_REP2.c_curve.txt:md5,cf4743abdd355595d6ec1fb3f38e66e5",
                "jurkat_REP2.lc_extrap.txt:md5,8ae0a77bcc59da38ec2da0fc1f973227"
            ],
            [
                "cd4_REP1.pos.DupRate.xls:md5,a80db2d20096ca839a7847ec5b11bf75",
                "cd4_REP1.seq.DupRate.xls:md5,c34531fd7578c6f62cbad53b96a7feb9",
                "cd4_REP2.pos.DupRate.xls:md5,06200ab67a60bee71fd168de88c15369",
                "cd4_REP2.seq.DupRate.xls:md5,c82f6d687eacabbab045db34647c3254",
                "cd4_REP3.pos.DupRate.xls:md5,fd3e91bfe070deccef65b265957cb733",
                "cd4_REP3.seq.DupRate.xls:md5,f4c9f61081e6ef309abe82bb34aa4bd0",
                "cd4_REP4.pos.DupRate.xls:md5,7fb8a7e96fd521561f62ccf7bf490212",
                "cd4_REP4.seq.DupRate.xls:md5,8c671e5b0029d396a666839029298828",
                "jurkat_REP1.pos.DupRate.xls:md5,0721c91ab7c640b046689095047657f8",
                "jurkat_REP1.seq.DupRate.xls:md5,aba941b1bf0e93f99e39bd507d1c02de",
                "jurkat_REP2.pos.DupRate.xls:md5,253ec69be4f5322dc92f6cb3fb41f7e0",
                "jurkat_REP2.seq.DupRate.xls:md5,c7c83a29e6dfd49ff17858fe83dc07c8"
            ],
            [
                "cd4_REP1.DupRate_plot.r:md5,a6f96b5b87a142dca2e09868deb8222b",
                "cd4_REP2.DupRate_plot.r:md5,a0686d22ba07f33a627c1a106d442a03",
                "cd4_REP3.DupRate_plot.r:md5,5b37293c52beaecc5980aa2b9bafd82c",
                "cd4_REP4.DupRate_plot.r:md5,ffa0f611b7f86d36134f9d80c3e1c9de",
                "jurkat_REP1.DupRate_plot.r:md5,8e4dc99cae73c94eda275aeee08a5390",
                "jurkat_REP2.DupRate_plot.r:md5,3507444f5965ca98a448ab15659ba34f"
            ],
            [
                "cd4_REP1.infer_experiment.txt:md5,2d8f31fe48cfc1db9d4c46ba59c2c7f3",
                "cd4_REP2.infer_experiment.txt:md5,70ac3bf0d0c50d644ab97e1698021e55",
                "cd4_REP3.infer_experiment.txt:md5,b0157d081d0ba0c27b370915f13c8c8a",
                "cd4_REP4.infer_experiment.txt:md5,37ec4ee4ed5f707f073be5a7d4c9acfb",
                "jurkat_REP1.infer_experiment.txt:md5,3ec36bed252146eb9cdb68db30c1855e",
                "jurkat_REP2.infer_experiment.txt:md5,cde67a32fc922fb61fe5c5ea4b0e5c92"
            ],
            [
                "cd4_REP1.read_distribution.txt:md5,1fcc6afbb63242818d446b877a832c3a",
                "cd4_REP2.read_distribution.txt:md5,c2762d927c1c12d520ebee8160561189",
                "cd4_REP3.read_distribution.txt:md5,2f05c4cc866b4f8068e02008cad20cb3",
                "cd4_REP4.read_distribution.txt:md5,ececae7f535b4fd5606bf763fdba7a34",
                "jurkat_REP1.read_distribution.txt:md5,6c02a757ed379a6c19f6855e38b5d909",
                "jurkat_REP2.read_distribution.txt:md5,a5f8bed2cff9d780a31c3ce43e289d9e"
            ],
            [
                "cd4_REP1.dreg.bedGraph:md5,8948a8fa86d8f6d413b77983189ff56e",
                "cd4_REP1.minus.bedGraph:md5,2a1c34f9d9ef9ff1b9da7874b9e3aaad",
                "cd4_REP1.minus.bigWig:md5,5280319275c98dcce023779fa389884d",
                "cd4_REP1.plus.bedGraph:md5,1509ec3a921e3109c5914e1bcef8cf33",
                "cd4_REP1.plus.bigWig:md5,72ccab3173f2018a22a4b36841247ba2",
                "cd4_REP2.dreg.bedGraph:md5,29f865a5668fae4a52a41589ac2b3179",
                "cd4_REP2.minus.bedGraph:md5,c530bc34fa3ec7ac49e88ff65f9c2f92",
                "cd4_REP2.minus.bigWig:md5,5e748c794e037f441741f7f409c8c5ad",
                "cd4_REP2.plus.bedGraph:md5,a675141da2874ec08d91591e5ea8242b",
                "cd4_REP2.plus.bigWig:md5,08674d52e9eeb08807c33ed3e4b3d504",
                "cd4_REP3.dreg.bedGraph:md5,201e690c619ee231416b643740141588",
                "cd4_REP3.minus.bedGraph:md5,1c44c165f663d08c869a013b78f051b0",
                "cd4_REP3.minus.bigWig:md5,aa3f5e86c635e182f9173543751cfbfd",
                "cd4_REP3.plus.bedGraph:md5,a475098ed14da6f3242c4cfdf3abb713",
                "cd4_REP3.plus.bigWig:md5,b255ae7c47424c223c537d00ecc025b5",
                "cd4_REP4.dreg.bedGraph:md5,dfcfce7df666c0e24a3d0d4e21447a0b",
                "cd4_REP4.minus.bedGraph:md5,b7a3d3c822155dfd8052c67a330fd6d5",
                "cd4_REP4.minus.bigWig:md5,fc61c62613dc0f74a65c32419aba8f76",
                "cd4_REP4.plus.bedGraph:md5,5623367b681562a2567afc62a8da4151",
                "cd4_REP4.plus.bigWig:md5,ed00ad7a22c49860debc86cfa7ffb457",
                "jurkat_REP1.dreg.bedGraph:md5,c25a4fb095e9f7d6766a3ce33e08f7d8",
                "jurkat_REP1.minus.bedGraph:md5,8d5d9a41df6eb6c56b1bfd3f39dc1fc6",
                "jurkat_REP1.minus.bigWig:md5,d06c8015c996bf520fff17266fd01f84",
                "jurkat_REP1.plus.bedGraph:md5,6ed63e5983edaa74fb3965676efdb674",
                "jurkat_REP1.plus.bigWig:md5,7a02334f2c7300ffdb5a2253c0937390",
                "jurkat_REP2.dreg.bedGraph:md5,aad861b6fd8d296fe6bb4bc82d5eeeed",
                "jurkat_REP2.minus.bedGraph:md5,cd83486cb6bb71e6b6e31a69c94d0d85",
                "jurkat_REP2.minus.bigWig:md5,686f719978de948f7045acc95f9d1c82",
                "jurkat_REP2.plus.bedGraph:md5,569e8a25c9cf48d1ef953f3da77c48a1",
                "jurkat_REP2.plus.bigWig:md5,cbe3fd38fe8f9508645bbfad83924212"
            ],
            "cd4.bed:md5,ab94918610a560772fdbec591200295f",
            "jurkat.bed:md5,41908723fa423a78476feb2e7627a3a4",
            [
                "cd4_intersect.bed:md5,c45718bf5c21eb28b810d4800296728b",
                "jurkat_intersect.bed:md5,d41d8cd98f00b204e9800998ecf8427e"
            ],
            [
                "cd4_filtered.bed:md5,9765d778862a20feefb94d407773bef3",
                "jurkat_filtered.bed:md5,cae11a1bfb707ea2df5fe612ae7268c8"
            ],
            true
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-11-08T19:59:26.10702"
    },
    "Should work with gzipped references": {
        "content": [
            55
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-11-08T20:00:53.869213"
    },
    "software_versions": {
        "content": [
            {
                "BBMAP_PILEUP": {
                    "bbmap": 39.01,
                    "samtools": "1.16.1",
                    "pigz": 2.6
                },
                "BEDTOOLS_GENOMECOV_MINUS": {
                    "bedtools": "2.31.1"
                },
                "BEDTOOLS_GENOMECOV_PLUS": {
                    "bedtools": "2.31.1"
                },
                "BEDTOOLS_INTERSECT": {
                    "bedtools": "2.31.1"
                },
                "BEDTOOLS_INTERSECT_FILTER": {
                    "bedtools": "2.31.1"
                },
                "BEDTOOLS_MERGE": {
                    "bedtools": "2.31.1"
                },
                "BEDTOOLS_SORT": {
                    "bedtools": "2.31.1"
                },
                "BWA_INDEX": {
                    "bwa": "0.7.18-r1243-dirty"
                },
                "BWA_MEM": {
                    "bwa": "0.7.18-r1243-dirty",
                    "samtools": 1.2
                },
                "CAT_CAT": {
                    "pigz": "2.3.4"
                },
                "CUSTOM_GETCHROMSIZES": {
                    "getchromsizes": 1.2
                },
                "DEEPTOOLS_BAMCOVERAGE_MINUS": {
                    "deeptools": "3.5.1"
                },
                "DEEPTOOLS_BAMCOVERAGE_PLUS": {
                    "deeptools": "3.5.1"
                },
                "FASTP": {
                    "fastp": "0.23.4"
                },
                "FASTQC": {
                    "fastqc": "0.12.1"
                },
                "GTF2BED": {
                    "perl": "5.26.2"
                },
                "HOMER_MAKETAGDIRECTORY": {
                    "homer": 4.11,
                    "samtools": 1.11
                },
                "PINTS_CALLER": {
                    "python": "3.12.6",
                    "pints": "1.1.13"
                },
                "PRESEQ_CCURVE": {
                    "preseq": "3.1.1"
                },
                "PRESEQ_LCEXTRAP": {
                    "preseq": "3.1.1"
                },
                "RSEQC_INFEREXPERIMENT": {
                    "rseqc": "5.0.2"
                },
                "RSEQC_READDISTRIBUTION": {
                    "rseqc": "5.0.2"
                },
                "RSEQC_READDUPLICATION": {
                    "rseqc": "5.0.2"
                },
                "SAMTOOLS_MERGE": {
                    "samtools": 1.21
                },
                "SUBREAD_FEATURECOUNTS_GENE": {
                    "subread": "2.0.1"
                },
                "SUBREAD_FEATURECOUNTS_PREDICTED": {
                    "subread": "2.0.1"
                },
                "Workflow": {
                    "nf-core/nascent": "v2.3.0dev"
                }
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-11-08T19:59:25.439456"
    },
    "Should work with BWA Index": {
        "content": [
            59
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-08-03T07:38:19.100788775"
    }
}